From: Kei Tokunaga <ktokunag@redhat.com>
Subject: [RHEL5 PATCH 5/5] pciehp: 'Trying to free already-free IRQ XX' is  reported
Date: Wed, 29 Nov 2006 12:41:48 -0500
Bugzilla: 216940
Message-Id: <456DC65C.5080503@redhat.com>
Changelog: pciehp: free_irq called twice


BZ216940
https://bugzilla.redhat.com/bugzilla/show_bug.cgi?id=216940

This fix was posted to the community, and is waiting for
being merged to upstream.  It already got "Signed-off-by:"
from the maintainer.

This patch fixes the problem that the following error message
is reported when pciehp driver is rmmoded.

	Trying to free already-free IRQ XX

The cause is that pciehp driver is doing unnecessary 2nd
free_irq at driver unloading.  This patch removes this call.

This patch applies to -2.6.18-1.2747.el5 and I have verified
that it worked all right.

Thanks,
Kei

Signed-off-by: Kenji Kaneshige <kaneshige.kenji@jp.fujitsu.com>
Signed-off-by: Kristen Carlson Accardi <kristen.c.accardi@intel.com>

---
 drivers/pci/hotplug/pciehp_core.c |    7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

--- gregkh-2.6.orig/drivers/pci/hotplug/pciehp_core.c
+++ gregkh-2.6/drivers/pci/hotplug/pciehp_core.c
@@ -521,14 +521,9 @@ static void __exit unload_pciehpd(void)
 
 }
 
-static int hpdriver_context = 0;
-
 static void pciehp_remove (struct pcie_device *device)
 {
-	printk("%s ENTRY\n", __FUNCTION__);	
-	printk("%s -> Call free_irq for irq = %d\n",  
-		__FUNCTION__, device->irq);
-	free_irq(device->irq, &hpdriver_context);
+	/* XXX - Needs to be adapted to device driver model */
 }
 
 #ifdef CONFIG_PM

--

